extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -113,4 +110,32 @@ public void close()
     unions.clear();
   }
 
+  @Override
+  public void relocate(int oldPosition, int newPosition, ByteBuffer newBuffer)
+  {
+    NativeMemory nm = getNativeMemory(newBuffer);
+
+    Memory mem = new MemoryRegion(nm, newPosition, maxIntermediateSize);
+    Union newUnion = (Union) SetOperation.wrap(mem);
+
+    Union union = unions.get(oldPosition);
+    if (union != null) {
+      unions.remove(oldPosition);
+      unions.put(newPosition, newUnion);
+    } else {
+      unions.put(newPosition, newUnion);
+    }
+
+  }
+
+  private NativeMemory getNativeMemory(ByteBuffer buffer)
+  {
+    NativeMemory nm = nmCache.get(buffer);
```

Comment:
It's not a huge improvement, but it _is_ an improvement by any metric - computeIfAbsent is clearer, shorter, safer, more efficient and the "right" way to perform this kind of task in Java 8+ world. I don't see reasons not to use it in completely new code.